### PR TITLE
planning: (1)pull over more (2) tests

### PR DESCRIPTION
**Pull request by jmtao**

### CODE REVIEW

File: modules/planning/conf/traffic_rule_config.pb.txt [hunk 1]
```diff
@@ -63,6 +63,16 @@ config: {
   }
 }
 
+config: {
+  rule_id: PULL_OVER
+  enabled: true
+  pull_over {
+    max_stop_deceleration: 2.0
+    stop_distance: 0.5
```
Comment (jilinzhou) [inline modules/planning/conf/traffic_rule_config.pb.txt]: Does it mean the ego vehicle stops at 0.5 meters before the destination ENU point?

File: modules/planning/tasks/traffic_decider/pull_over.cc [hunk 2]
```diff
@@ -44,20 +44,20 @@ Status PullOver::ApplyRule(Frame* const frame,
...
 
   PointENU stop_point;
   double stop_heading = 0.0;
-  if (!GetPullOverStop(&stop_point, &stop_heading)) {
-    ADEBUG << "Could not find a safe pull over point";
-    return Status::OK();
-  }
-  if (!BuildPullOverStop(stop_point, stop_heading)) {
-    AWARN << "Not able to create a pull over stop";
+  if (GetPullOverStop(&stop_point, &stop_heading) < 0) {
```
Comment (jilinzhou) [inline modules/planning/tasks/traffic_decider/pull_over.cc]: if (GetPullOverStop(&stop_point, &stop_heading) != 0)

File: modules/planning/tasks/traffic_decider/pull_over.cc [hunk 3]
```diff
@@ -73,37 +73,61 @@ bool PullOver::IsValidStop(const PointENU& stop_point,
...
+
+  const auto& reference_line = reference_line_info_->reference_line();
+  double heading = reference_line.GetReferencePoint(stop_s).heading();
+  double lane_left_width = 0.0;
+  double lane_right_width = 0.0;
+  reference_line.GetLaneWidth(stop_s, &lane_left_width, &lane_right_width);
+  stop_point_sl.set_l(-lane_right_width);
+
+  common::math::Vec2d point;
+  reference_line.SLToXY(stop_point_sl, &point);
```
Comment (jilinzhou) [inline modules/planning/tasks/traffic_decider/pull_over.cc]: What happen if stop_point_sl.s() > reference_line.length()?

File: modules/planning/tasks/traffic_decider/pull_over.cc [hunk 4]
```diff
@@ -148,31 +172,31 @@ bool PullOver::CheckPullOver() {
...
 }
 
-bool PullOver::BuildPullOverStop(const PointENU& stop_point,
+int PullOver::BuildPullOverStop(const PointENU& stop_point,
                                  double stop_heading) {
   // check
   const auto& reference_line = reference_line_info_->reference_line();
   common::SLPoint sl;
   reference_line.XYToSL(stop_point, &sl);
   if (sl.s() < 0 || sl.s() > reference_line.Length()) {
```
Comment (jilinzhou) [inline modules/planning/tasks/traffic_decider/pull_over.cc]: Is it necessary to double check sl.s() here?

File: modules/planning/conf/traffic_rule_config.pb.txt (hunk 1 above)
Comment (jmtao) [reply modules/planning/conf/traffic_rule_config.pb.txt]: yes.  it's the distance adc keeps from virtual stop wall when adc stops.

File: modules/planning/tasks/traffic_decider/pull_over.cc (hunk 3 above)
Comment (jmtao) [reply modules/planning/tasks/traffic_decider/pull_over.cc]: stop_point is adc's position + plan_distance(config, currently it is set to 20m). It shall be within reference line.

File: modules/planning/tasks/traffic_decider/pull_over.cc (hunk 4 above)
Comment (jmtao) [reply modules/planning/tasks/traffic_decider/pull_over.cc]: it is always recommended to add checks. you can't assume the parameters passed in are all legit.

File: modules/planning/tasks/traffic_decider/pull_over.cc (hunk 2 above)
Comment (jmtao) [reply modules/planning/tasks/traffic_decider/pull_over.cc]: either way

**Pull request merged**